app/jobs/bulk_update_job.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+class BulkUpdateJob < ApplicationJob
+  def perform(ids:, args:)
+    change_set_persister.buffer_into_index do |buffered_change_set_persister|
+      ids.each do |id|
+        resource = query_service.find_by(id: id)
+        change_set = DynamicChangeSet.new(resource).prepopulate!
+        attributes = {}.tap do |attrs|
+          attrs[:state] = "complete" if args[:mark_complete] && !resource.state.include?("complete")
+        end
+        change_set.validate(attributes)
+        if change_set.changed?
```

Comment:
I don't think this is going to do what you want. Should be `if change_set.validate(attributes)` or `if change_set.valid?`.

`change_set.changed?` will just tell you if a property is different, which it really should be if it's being bulk updated.

I'm actually a little concerned the tests below pass in light of this.

Reply:
OH, I see what's going on. Ignore me.